state_test.go

```diff
@@ -306,9 +303,10 @@ func TestStateUpdateReportStatus(t *testing.T) {
 	s, c = usr.Handle(&ctx, sc)
 	assert.IsType(t, s, &ReportErrorState{})
 	assert.False(t, c)
-	assert.WithinDuration(t, time.Now(), now1, 3*time.Second)
-	assert.InDelta(t, int(maxReportSendingTime/poll),
-		usr.(*UpdateStatusReportState).triesSendingReport, 100)
+	assert.WithinDuration(t, time.Now(), now1,
+		time.Duration(maxReportSendingAttempts)*time.Millisecond+time.Second)
```

Comment:
Nope. This is to wait `maxReportSendingAttempts` multiplied by the time needed for one reqyest plus some buffer (1s in this case).